src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -73,15 +80,27 @@
      *             if serverAddresses is empty or resolves to an empty list
      */
     public StaticHostProvider(Collection<InetSocketAddress> serverAddresses) {
-       sourceOfRandomness = new Random(System.currentTimeMillis() ^ this.hashCode());
+        init(serverAddresses,
```

Comment:
Nit: Call StaticHostProvider(Collection<InetSocketAddress> serverAddresses, Resolver resolver).

Please change the other two constructor as well.

Reply:
I think what @lvfangmin meant is instead of duplicating the code inside three constructors, only keep one concrete parameterized constructor implementation and let the other two invoke that one (with different parameters.).

Reply:
What hanm said, instead of call init, call this.

Reply:
Tried that. Cannot call the other constructor, because `this` is referenced.

Reply:
Yeah, I think referencing this before the object is constructed is prohibited. I think we can probably get around it but it looks like not worth the extra effort (I don't know a simple of way of doing it without dropping using the this.hashCode). 
So I am ok with the current form of having all ctors invoking init method.